editor/editor_node.cpp

```diff
@@ -638,6 +638,32 @@ void EditorNode::_notification(int p_what) {
 					EditorFileSystem::get_singleton()->scan();
 				}
 			}
+
+			bool system_theme_changed = false;
+
+			bool follow_system_theme = EDITOR_GET("interface/theme/system_theme/follow_system_theme");
```

Comment:
Reading project settings every frame should be avoided for performance reasons. I suggest using the `editor_settings_changed` signal we have somewhere to store this value to a variable, and only read from that variable instead.

Also, I suggest not performing this check every frame, but rather every second or so to reduce resource usage. This can be done using a counter variable or a Timer node.